app/src/main/java/com/github/gotify/service/GotifyMessengerService.kt

```diff
@@ -0,0 +1,176 @@
+package com.github.gotify.service
+
+import android.app.Service
+import android.content.Intent
+import android.os.*
+import androidx.annotation.RequiresApi
+import androidx.core.os.bundleOf
+import com.github.gotify.Settings
+import com.github.gotify.api.ClientFactory
+import com.github.gotify.client.api.ApplicationApi
+import com.github.gotify.log.Log
+import java.io.IOException
+
+/**
+ * THIS SERVICE IS USED BY OTHER APPS TO REGISTER
+ */
+// TODO : in the app, implement forceUnregisterApp
+
+@RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+class GotifyMessengerService : Service() {
+    /** Keeps track of all current registered clients.  */
+    private val db = MessagingDatabase(this)
+    private lateinit var settings: Settings
+
+    /**
+     * Handler of incoming messages from clients.
+     */
+    internal inner class gHandler : Handler() {
+
+        override fun handleMessage(msg: Message) {
+            when (msg.what) {
+                MSG_START -> simpleAnswer(msg, MSG_START)
+                MSG_REGISTER_CLIENT -> {
+                    val uid = msg.sendingUid
+                    val msgData = msg.data
+                    val thread = object : Thread() {
+                        override fun run() {
+                            registerApp(msgData, uid)
+                        }
+                    }
+                    thread.start()
+                    // We need to wait until the registration is done to send url & token :
+                    thread.join()
```

Comment:
Kotlin has a helper function for creating threads: kotlin.concurrent.ThreadsKt#thread
```suggestion
                    val uid = msg.sendingUid
                    val msgData = msg.data
                    thread(start = true) {
                        registerApp(msgData, uid)
                    }.join()
```
this should be used for the other usages as well.

Reply:
done